plugin/pkg/scheduler/algorithm/priorities/taint_toleration.go

```diff
@@ -43,40 +42,33 @@ func countIntolerableTaintsPreferNoSchedule(taints []v1.Taint, tolerations []v1.
 
 // getAllTolerationEffectPreferNoSchedule gets the list of all Toleration with Effect PreferNoSchedule
 func getAllTolerationPreferNoSchedule(tolerations []v1.Toleration) (tolerationList []v1.Toleration) {
-	for i := range tolerations {
-		toleration := &tolerations[i]
-		if len(toleration.Effect) == 0 || toleration.Effect == v1.TaintEffectPreferNoSchedule {
```

Comment:
I thought we don't need it as we are comparing with constants, so if it is needed probably && would have helped as for short circuit evaluation.